### PR TITLE
Create periodic ECR scan

### PR DESCRIPTION
**Pull request by MancunianSam**

### CODE REVIEW

File: cloudwatch_events/outputs.tf [hunk 1]
```diff
@@ -0,0 +1,3 @@
+output "event_arn" {
+  value = local.event_rule_arn
+}
```
Comment (TomJKing) [inline cloudwatch_events/outputs.tf:3]: Missing line break

File: lambda/ecr_scan_notifications.tf [hunk 2]
```diff
@@ -48,3 +48,12 @@ resource "aws_iam_role_policy_attachment" "ecr_scan_notifications_lambda_role_po
   policy_arn = aws_iam_policy.ecr_scan_notifications_lambda_policy.*.arn[0]
   role       = aws_iam_role.ecr_scan_notifications_lambda_iam_role.*.name[0]
 }
+
+resource "aws_lambda_permission" "lambda_permissions" {
```
Comment (TomJKing) [inline lambda/ecr_scan_notifications.tf:52]: Has  `terraform fmt` been run on this file?

File: lambda/templates/ecr_scan_lambda.json.tpl [hunk 3]
```diff
@@ -0,0 +1,26 @@
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Effect": "Allow",
+      "Action": [
+        "ecr:StartImageScan",
```
Comment (TomJKing) [inline lambda/templates/ecr_scan_lambda.json.tpl:7]: Could you order the `Action`s alphabetically?

File: lambda/templates/ecr_scan_lambda.json.tpl [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+        "logs:CreateLogStream",
+        "logs:PutLogEvents"
+      ],
+      "Resource": [
+        "arn:aws:logs:eu-west-2:${account_id}:log-group:/aws/lambda/tdr-ecr-scan-${environment}",
+        "arn:aws:logs:eu-west-2:${account_id}:log-group:/aws/lambda/tdr-ecr-scan-${environment}:log-stream:*"
+      ]
+    }
+  ]
+}
```
Comment (TomJKing) [inline lambda/templates/ecr_scan_lambda.json.tpl:26]: Missing line break

File: lambda/variables.tf [hunk 5]
```diff
@@ -102,3 +107,12 @@ variable "mount_target_ids" {
...
 }
+
+variable "event_rule_arns" {
+  type    = set(string)
+  default = []
+}
+
+variable "periodic_ecr_image_scan_event_arn" {
+  default = ""
+}
```
Comment (TomJKing) [inline lambda/variables.tf:118]: Missing line breaks

**Review by TomJKing — COMMENTED** [review]
Looks good. Some very minor comments.

File: cloudwatch_events/outputs.tf (hunk 1 above)
Comment (MancunianSam) [reply cloudwatch_events/outputs.tf:3]: I've added it in.

File: lambda/ecr_scan_notifications.tf (hunk 2 above)
Comment (MancunianSam) [reply lambda/ecr_scan_notifications.tf:52]: No, sorry.

File: lambda/templates/ecr_scan_lambda.json.tpl (hunk 3 above)
Comment (MancunianSam) [reply lambda/templates/ecr_scan_lambda.json.tpl:7]: Yep, I've sorted them.

File: lambda/variables.tf (hunk 5 above)
Comment (MancunianSam) [reply lambda/variables.tf:118]: I've added this one and the one above.

**Review by TomJKing — APPROVED** [review]

**Pull request merged**